// node-lucene/dist/src/index/IndexWriterConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const node_java_rt_1 = require("node-java-rt");
const getLuceneJava_1 = require("../util/getLuceneJava");
class IndexWriterConfig extends node_java_rt_1.lang.Object {
    constructor(analyzer) {
        super();
        this._java = getLuceneJava_1.getLuceneJava().newInstanceSync(IndexWriterConfig._javaClassName(), analyzer._java);
    }
    static _javaClassName() {
        return 'org.apache.lucene.index.IndexWriterConfig';
    }
    getRAMBufferSizeMBSync() {
        return this._java.getRAMBufferSizeMBSync();
    }
    getRAMBufferSizeMB() {
        return this._java.getRAMBufferSizeMBSync();
    }
    getRAMBufferSizeMBAsync(callback) {
        this._java.getRAMBufferSizeMBAsync(callback);
    }
    getRAMBufferSizeMBPromise() {
        return this._java.getRAMBufferSizeMBPromise();
    }
    setOpenMode(openMode) {
        return this._java.setOpenModeSync(openMode._java);
    }
    setOpenModeSync(openMode) {
        return this._java.setOpenModeSync(openMode._java);
    }
    setOpenModePromise(openMode) {
        return this._java.setOpenModePromise(openMode._java);
    }
}
exports.default = IndexWriterConfig;
