// node-lucene/dist/src/analysis/standard/StandardAnalyzer.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getLuceneJava_1 = require("../../util/getLuceneJava");
const Analyzer_1 = __importDefault(require("../Analyzer"));
const TokenStream_1 = __importDefault(require("../TokenStream"));
class StandardAnalyzer extends Analyzer_1.default {
    constructor() {
        super();
        this._java = getLuceneJava_1.getLuceneJava().newInstanceSync(StandardAnalyzer._javaClassName());
    }
    static _javaClassName() {
        return 'org.apache.lucene.analysis.standard.StandardAnalyzer';
    }
    getMaxTokenLengthSync() {
        return this._java.getMaxTokenLengthSync();
    }
    getMaxTokenLength() {
        return this._java.getMaxTokenLengthSync();
    }
    getMaxTokenLengthAsync(callback) {
        this._java.getMaxTokenLengthAsync(callback);
    }
    getMaxTokenLengthPromise() {
        return this._java.getMaxTokenLengthPromise();
    }
    close() {
        this._java.closeSync();
    }
    tokenStream(fieldName, text) {
        return StandardAnalyzer._buildSyncOrThrow(this._java.tokenStreamSync(fieldName, text), new TokenStream_1.default());
    }
}
exports.default = StandardAnalyzer;
