// node-java-rt/dist/src/java/index.js
"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./java"));
__export(require("./JavaBase"));
// java.lang
const Object_1 = __importDefault(require("./lang/Object"));
const Enum_1 = __importDefault(require("./lang/Enum"));
const Runtime_1 = __importDefault(require("./lang/Runtime"));
const ClassLoader_1 = __importDefault(require("./lang/ClassLoader"));
const Class_1 = __importDefault(require("./lang/Class"));
const Double_1 = __importDefault(require("./lang/Double"));
const Proxy_1 = __importDefault(require("./lang/reflect/Proxy"));
const Method_1 = __importDefault(require("./lang/reflect/Method"));
exports.lang = {
    Double: Double_1.default,
    Object: Object_1.default,
    ClassLoader: ClassLoader_1.default,
    Class: Class_1.default,
    Enum: Enum_1.default,
    Runtime: Runtime_1.default,
    reflect: {
        // InvocationHandler: InvocationHandlerT,
        Method: Method_1.default,
        Proxy: Proxy_1.default
    }
};
// java.util
const LinkedList_1 = __importDefault(require("./util/LinkedList"));
const EventObject_1 = __importDefault(require("./util/EventObject"));
exports.util = {
    LinkedList: LinkedList_1.default,
    EventObject: EventObject_1.default
};
exports.io = {};
// java.nio
const Paths_1 = __importDefault(require("./nio/file/Paths"));
exports.nio = {
    file: {
        Paths: Paths_1.default
    }
};
// JavaBase.math
const BigInteger_1 = __importDefault(require("./math/BigInteger"));
exports.math = {
    BigInteger: BigInteger_1.default
};
