// node-java-rt/dist/src/java/java.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
let java;
let defaultJavaOptions = {
    asyncOptions: {
        asyncSuffix: 'Async',
        syncSuffix: 'Sync',
        promiseSuffix: 'Promise',
        promisify: require('util').promisify
    },
    classpath: [],
    options: []
};
function setJavaOptions(options) {
    defaultJavaOptions = options; //TODO: Object.assign preserve defaults recursively
}
exports.setJavaOptions = setJavaOptions;
function getJava() {
    if (!java) {
        java = require('java');
        // Object.assign(java, javaOptions) // TODO: throw native v8 exception ! - report to node-java ?
        java.asyncOptions = defaultJavaOptions.asyncOptions;
        defaultJavaOptions.classpath.forEach(cp => java.classpath.push(cp));
        java.options = defaultJavaOptions.options;
    }
    return java;
}
exports.getJava = getJava;
function getJavaObjectOrThrow(obj) {
    if (obj._java) {
        return obj._java;
    }
    else {
        throw new Error(`Object ${obj} expected to have property _java`);
    }
}
exports.getJavaObjectOrThrow = getJavaObjectOrThrow;
